stm32-modules/heater-shaker/firmware/system/system_serial_number.h

```diff
@@ -8,7 +8,7 @@ extern "C" {
 #include <stdbool.h>
 #include <stddef.h>
 
-bool system_set_serial_number(uint64_t to_write, uint8_t address);
+bool system_set_serial_number(uint64_t *to_write);
```

Comment:
since we're not really constrained on code speed or size, can we make this more like

```c
struct writable_serial {
   uint64_t contents[3];
};

bool system_set_serial_number(struct writable_serial to_write);
// or
bool system_set_serial_number(struct writable_serial* to_write);
``` 

that plays a bit better with static analysis since it's now a bounded array; it's in a struct so it doesn't [decay](https://stackoverflow.com/questions/1461432/what-is-array-to-pointer-decay)